### PR TITLE
List tables

### PR DESCRIPTION
**Pull request by rwhelan**
This PR adds `ListTables()` to retrieve currently configured tables within the kernel.

This PR builds on PR#15

### CODE REVIEW

**Comment by stapelberg** [comment]
Could you rebase this please?

**Comment by RyanAtCNG** [comment]
Of Course!

Other List functions my be on the horizon as my project matures.

Thanks!

**Comment by rwhelan** [comment]
Of Course!

More List functions maybe on the way as my project matures.

Thanks for the library; very helpful!

File: table.go [hunk 1]
```diff
@@ -66,3 +71,66 @@ func (cc *Conn) AddTable(t *Table) *Table {
...
+	ad, err := netlink.NewAttributeDecoder(msg.Data[4:])
+	if err != nil {
+		return nil, err
+	}
+
+	for ad.Next() {
+		switch ad.Type() {
+		case unix.NFTA_TABLE_NAME:
+			t.Name = ad.String()
+			/*
```
Comment (stapelberg) [inline table.go]: What about this commented out code? Is there anything left to do here (if so, can you implement it?) or is this cruft?

Comment (rwhelan) [reply table.go]: They are attributes returned from the kernel that are not yet represented on the Table struct.  I found them when decoding kernel responses- but don't yet have a need for them.  I put them (commented out) on the decoding function as a reminder but forgot they were there when submitting the PR.

I will remove them for now; unless you'd prefer to extend the structs to represent the attributes for completeness?

Comment (stapelberg) [reply table.go]: If they’re easy enough to test, let’s do completeness to save the person needing them some work :) 

If not, you can leave them out.

**Comment by rwhelan** [comment]
I added 2 of the struct members to the decoding and everything looks to be working.  I contemplated created an enum for the `Flags` values, but looking at the header file on my machine, it looks like there is only a single flag defined...

**Comment by stapelberg** [comment]
Thanks!

**Pull request merged**